### PR TITLE
fl history --chronological, fl history FRIEND_NAME

### PR DESCRIPTION
**Pull request by annapapitto**
Addresses https://github.com/marwahaha/friendlog/issues/22

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -220,23 +220,38 @@ function fail(msg) {
...
-  Object.keys(eventsByFriend).sort().forEach(friend => {
-    if (!friendName || friendName === friend) {
-      prettyPrintFriendHeader(friend);
-      _.sortBy(eventsByFriend[friend], "date").reverse().map(prettyPrintEvent);
-      console.log();
-    }
-  });
+  var perFriend = function(friend) {
+    prettyPrintFriendHeader(friend);
+    _.sortBy(eventsByFriend[friend], "date").reverse().map(prettyPrintEventNameless);
```
Comment (marwahaha) [inline index.js]: What changed here?

File: index.js [hunk 2]
```diff
@@ -220,23 +220,38 @@ function fail(msg) {
 }
 
 function showHistory(friendName) {
+  if (friendName == "--chronological") {
+    return showChronologicalHistory();
```
Comment (marwahaha) [inline index.js]: Should we ban friends that start with `--` ? (This feels like a hack :frowning_face: )

File: index.js (hunk 1 above)
Comment (prendradjaja) [reply index.js]: Kunal were you looking at the full diff (not the diff for the individual commit)? The diff for the commit looks pretty clear.

File: index.js (hunk 2 above)
Comment (prendradjaja) [reply index.js]: +1, it's probably better to do some proper argument-parsing here.

File: index.js [hunk 3]
```diff
@@ -222,13 +222,16 @@ function fail(msg) {
...
   var events = loadEventsData();
   var eventsByFriend = _.groupBy(events, "name");
-  Object.keys(eventsByFriend).sort().forEach(friend => {
-    if (!friendName || friendName === friend) {
-      prettyPrintFriendHeader(friend);
-      _.sortBy(eventsByFriend[friend], "date").reverse().map(prettyPrintEvent);
-      console.log();
-    }
-  });
+  var perFriend = function(friend) {
```
Comment (prendradjaja) [inline index.js]: I think this should be extracted into a top-level function, call it `showHistoryFor(friend)` or `showHistoryForFriend(friend)` perhaps.

**Comment by marwahaha** [comment]
I merged these commits into master and made some light changes.

**Pull request closed without merging**